### ISSUE TITLE
Loss value when training QueryInst

### ISSUE THREAD

**Issue by bnbao** [open-mmlab/mmdetection#9073]
### Prerequisite

- [X] I have searched [the existing and past issues](https://github.com/open-mmlab/mmdetection/issues) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmdetection).

### 💬 Describe the reimplementation questions

I have followed the tutorial on how to train QueryInst with custom data and modified config and other files accordingly to include my class names and number of classes, and have also organised my data into the coco format. But, when I try to train, I get this message displayed recursively a hundred times for every epoch, "Ground Truth Not Found!" - with the original repo from author.

### Environment

GPU A6000
mmcv:1.3.17
pytorch: 1.10.1 cuda 11.3

### Expected results

 I tried to train on your work so the loss values zero except loss_cls. Can you help me to solve that, I really appreciate it.

[stage0_loss_cls: 0.0025, stage0_pos_acc0, stage4_loss_iou: 0.0000, stage4_loss_mask: 0.0000, stage5_loss_cls: 0.0024, stage5_pos_acc: 0.0000, stage5_loss_bbox: 0.0000, stage5_loss_iou: 0.0000, stage5_loss_mask: 0.0000,0.0000, stage1_loss_mask: 0.0000, stage2_loss_cls: 0.0012, stage2_pos_acc: 0.0000, stage2_loss_bbox: 0.0000, stage2_loss_iou: 0.0000, stage2_loss_mask: 0.0000, stage3_loss_cls: 0.0012, stage3_pos_acc: 0.0000, stage3_loss_bbox: 0.0000, stage3_loss_iou: 0.0000, stage3_loss_mask: 0.0000, stage4_loss_cls: 0.0010, stage4_pos_acc: 0.0000, stage4_loss_bbox: 0.000](lr: 1.499e-05, eta: 17:44:38, time: 0.620, data_time: 0.004, memory: 10924, stage0_loss_cls: 125712.0807, stage0_pos_acc: 0.0000, stage0_loss_bbox: 0.0000, stage0_loss_iou: 0.0000, stage0_loss_mask: 0.0000, stage1_loss_cls: 90232.6852, stage1_pos_acc: 0.0000, stage1_loss_bbox: 0.0000, stage1_loss_iou: 0.0000, stage1_loss_mask: 0.0000, stage2_loss_cls: 46718.8418, stage2_pos_acc: 0.0000, stage2_loss_bbox: 0.0000, stage2_loss_iou: 0.0000, stage2_loss_mask: 0.0000, stage3_loss_cls: 32931.4105, stage3_pos_acc: 0.0000, stage3_loss_bbox: 0.0000, stage3_loss_iou: 0.0000, stage3_loss_mask: 0.0000, stage4_loss_cls: 41905.0223, stage4_pos_acc: 0.0000, stage4_loss_bbox: 0.0000, stage4_loss_iou: 0.0000, stage4_loss_mask: 0.0000, stage5_loss_cls: 47268.3891, stage5_pos_acc: 0.0000, stage5_loss_bbox: 0.0000, stage5_loss_iou: 0.0000, stage5_loss_mask: 0.0000, loss: 384768.4313, grad_norm: 1482184.3400)

**Comment by wanghonglie** [comment]
> Hi, maybe you can print results inside the `__call__` method of pipeline `LoadAnnotations` in file `mmdet/datasets/pipelines/loading.py` to see if the labels is loaded correctly.

**Comment by ZwwWayne** [comment]
It seems that there is no ground truth boxes or masks are used to train the model, therefore, there are many backgrounds and only loss_cls is not 0.

**Comment by alexwangxiang** [comment]
I'm also facing this problem, classification loss often explodes during the training process, but bbox and mask loss seem to work correctly. Here is my training log:

2023-09-28 10:03:39,574 - mmdet - INFO - Epoch [1][200/587]	lr: 1.998e-06, eta: 2:31:32, time: 0.434, data_time: 0.041, memory: 5048, stage0_loss_cls: 1898.9996, stage0_pos_acc: 0.1176, stage0_loss_bbox: 0.8629, stage0_loss_iou: 1.9623, stage0_loss_mask: 6.5378, stage1_loss_cls: 3114.6418, stage1_pos_acc: 0.2548, stage1_loss_bbox: 0.5850, stage1_loss_iou: 1.5873, stage1_loss_mask: 5.4482, stage2_loss_cls: 5812.7446, stage2_pos_acc: 3.6593, stage2_loss_bbox: 0.5564, stage2_loss_iou: 1.4486, stage2_loss_mask: 5.2258, stage3_loss_cls: 5565.6365, stage3_pos_acc: 2.2306, stage3_loss_bbox: 0.5340, stage3_loss_iou: 1.3920, stage3_loss_mask: 5.2454, stage4_loss_cls: 4104.3203, stage4_pos_acc: 0.5068, stage4_loss_bbox: 0.5231, stage4_loss_iou: 1.3718, stage4_loss_mask: 5.1587, stage5_loss_cls: 3669.5920, stage5_pos_acc: 0.7288, stage5_loss_bbox: 0.5163, stage5_loss_iou: 1.3680, stage5_loss_mask: 4.9600, loss: 24211.2203, grad_norm: 170524.9333
2023-09-28 10:04:59,918 - mmdet - INFO - Epoch [1][400/587]	lr: 3.996e-06, eta: 2:24:27, time: 0.402, data_time: 0.033, memory: 5048, stage0_loss_cls: 2.4376, stage0_pos_acc: 0.2046, stage0_loss_bbox: 0.7862, stage0_loss_iou: 1.9436, stage0_loss_mask: 6.2967, stage1_loss_cls: 2.1715, stage1_pos_acc: 1.1013, stage1_loss_bbox: 0.4365, stage1_loss_iou: 1.4621, stage1_loss_mask: 4.5922, stage2_loss_cls: 1.9628, stage2_pos_acc: 12.7806, stage2_loss_bbox: 0.3886, stage2_loss_iou: 1.2403, stage2_loss_mask: 4.3191, stage3_loss_cls: 2.1487, stage3_pos_acc: 2.2326, stage3_loss_bbox: 0.3685, stage3_loss_iou: 1.1627, stage3_loss_mask: 4.2869, stage4_loss_cls: 2.1877, stage4_pos_acc: 0.1646, stage4_loss_bbox: 0.3485, stage4_loss_iou: 1.1279, stage4_loss_mask: 4.2234, stage5_loss_cls: 2.0325, stage5_pos_acc: 2.6923, stage5_loss_bbox: 0.3421, stage5_loss_iou: 1.1148, stage5_loss_mask: 4.0324, loss: 51.4132, grad_norm: 142.5453
2023-09-28 10:07:30,713 - mmdet - INFO - Epoch [2][200/587]	lr: 7.862e-06, eta: 1:45:47, time: 0.392, data_time: 0.033, memory: 5319, stage0_loss_cls: 8927.1670, stage0_pos_acc: 0.0810, stage0_loss_bbox: 0.6170, stage0_loss_iou: 1.8565, stage0_loss_mask: 5.0327, stage1_loss_cls: 32954.9045, stage1_pos_acc: 38.8785, stage1_loss_bbox: 0.2778, stage1_loss_iou: 1.1677, stage1_loss_mask: 3.0968, stage2_loss_cls: 38723.2565, stage2_pos_acc: 43.1120, stage2_loss_bbox: 0.2346, stage2_loss_iou: 0.9330, stage2_loss_mask: 3.0297, stage3_loss_cls: 72593.0452, stage3_pos_acc: 52.4596, stage3_loss_bbox: 0.2199, stage3_loss_iou: 0.8660, stage3_loss_mask: 3.0169, stage4_loss_cls: 37119.1816, stage4_pos_acc: 48.2841, stage4_loss_bbox: 0.2089, stage4_loss_iou: 0.8426, stage4_loss_mask: 2.9683, stage5_loss_cls: 76735.3634, stage5_pos_acc: 58.2903, stage5_loss_bbox: 0.2035, stage5_loss_iou: 0.8251, stage5_loss_mask: 2.6969, loss: 267081.0225, grad_norm: 5634387.9443
2023-09-28 10:08:48,001 - mmdet - INFO - Epoch [2][400/587]	lr: 9.860e-06, eta: 1:49:48, time: 0.386, data_time: 0.024, memory: 5319, stage0_loss_cls: 32674.9770, stage0_pos_acc: 19.0476, stage0_loss_bbox: 0.5492, stage0_loss_iou: 1.8356, stage0_loss_mask: 4.1550, stage1_loss_cls: 161677.6105, stage1_pos_acc: 42.9056, stage1_loss_bbox: 0.2330, stage1_loss_iou: 1.0549, stage1_loss_mask: 2.5213, stage2_loss_cls: 115156.7964, stage2_pos_acc: 49.9250, stage2_loss_bbox: 0.1864, stage2_loss_iou: 0.8447, stage2_loss_mask: 2.4285, stage3_loss_cls: 118818.6955, stage3_pos_acc: 58.7734, stage3_loss_bbox: 0.1713, stage3_loss_iou: 0.7594, stage3_loss_mask: 2.4332, stage4_loss_cls: 85852.9543, stage4_pos_acc: 60.0680, stage4_loss_bbox: 0.1570, stage4_loss_iou: 0.7437, stage4_loss_mask: 2.4201, stage5_loss_cls: 99059.8253, stage5_pos_acc: 66.9799, stage5_loss_bbox: 0.1483, stage5_loss_iou: 0.7241, stage5_loss_mask: 2.1469, loss: 613264.3767, grad_norm: 12634433.8133

**Comment by alexwangxiang** [comment]
Finally I found out what the problem is. Previously I trained my model on a single 3090 ti gpu. Due to its lack of memory (24GB), I have to set samples_per_gpu=1, and workers_per_gpu=1. Then I trained my model on another computer with A100 gpu (80GB memory), and set samples_per_gpu=3, workers_per_gpu=3, the training loss appeared correct and never explodes again.
